### PR TITLE
Arrange to forward pg_basebackup output while it's running.

### PR DESCRIPTION
**Pull request by DimCitus**
This can be a pretty long operation and being in the dark for its whole
duration is not a great user experience.

### CODE REVIEW

File: src/bin/pg_autoctl/pgctl.c [hunk 1]
```diff
@@ -633,29 +639,45 @@ pg_basebackup(const char *pgdata,
...
+	args[argsIndex++] = "--wal-method=stream";
+	args[argsIndex++] = "--slot";
+	args[argsIndex++] = replicationSource->slotName;
+	args[argsIndex] = NULL;
 
-	/* pg_basebackup uses stderr for all of its output */
-	(void) log_program_output(program, LOG_INFO, LOG_INFO);
+	/*
+	 * We do not want to call setsid() when running this program, as the
+	 * postgres subprogram is not intended to be its own session leader, but
```
Comment (JelteF) [inline src/bin/pg_autoctl/pgctl.c:661]:
```suggestion
	 * pg_basebackup subprogram is not intended to be its own session leader, but
```

File: src/bin/pg_autoctl/string_utils.c [hunk 2]
```diff
@@ -443,3 +443,29 @@ splitLines(char *errorMessage, char **linesArray, int size)
...
+{
+	char *outLines[BUFSIZE] = { 0 };
+	int lineCount = splitLines((char *) buffer, outLines, BUFSIZE);
+	int lineNumber = 0;
+
+	for (lineNumber = 0; lineNumber < lineCount; lineNumber++)
+	{
+		/*
+		 * pg_basebackup and other utilities write their progress output on
+		 * stderr, we don't want to have ERROR message when it's all good.
```
Comment (JelteF) [inline src/bin/pg_autoctl/string_utils.c:464]: This comment seems to have nothing to do with the code below. It seems that simply ignores empty lines. It's good to note somewhere that processBufferCallback is called both for `stderr` and `stdout`, but this does not seem like the right place.

File: src/bin/pg_autoctl/pgctl.c (hunk 1 above)
Comment (DimCitus) [reply src/bin/pg_autoctl/pgctl.c]: Fixed.

File: src/bin/pg_autoctl/string_utils.c (hunk 2 above)
Comment (DimCitus) [reply src/bin/pg_autoctl/string_utils.c]: Ah yeah. Good point. Revised now, thanks!

**Pull request merged**